mlir/lib/Conversion/AIRToAIEPass.cpp

```diff
@@ -2290,6 +2290,8 @@ class AIRToAIEPass : public air::impl::AIRToAIEBase<AIRToAIEPass> {
       builder.setInsertionPoint(bco.getOperand().getDefiningOp());
     else if (auto a = dyn_cast<memref::AllocaOp>(alloc.getDefiningOp()))
       builder.setInsertionPoint(alloc.getDefiningOp());
+    else if (auto a = dyn_cast<AIE::BufferOp>(alloc.getDefiningOp()))
+      builder.setInsertionPointToStart(memcpyOpIf->getBlock());
```

Comment:
The fall-though path is to lock the buffer at the memcpy callsite, which is highly suspect (isn't that the point at which compute is _done_ with the buffer?). This probably isn't the right answer in general but it's better than the fall-through when this function runs after AIE::BufferOp allocation (which is what I'm seeing).

Reply:
Agreed. Makes sense to me.